cmd/common-main.go

```diff
@@ -123,3 +123,23 @@ func handleCommonEnvVars() {
 		globalIsEnvDomainName = true
 	}
 }
+
+func loadCerts() {
+	// Check for certificates only if endpoints are secure.
+	if ssl := globalEndpoints.IsHTTPS(); ssl {
+		// Check and load SSL certificates under distributed mode.
+		var err error
+		globalPublicCerts, globalRootCAs, globalTLSCertificate, err = getSSLConfig()
+
+		// Set globalIsSSL to true only if certs can be loaded properly.
+		globalIsSSL = err == nil
+
+		// When not running in distributed mode, for example in gateway we ignore
+		// if we cannot load certs, reset the error as success.
```

Comment:
Can't understand why.  Isn't it possible to run a insecure gateway with TLS backend? similarly secure gateway with insecure backend?

Reply:
Yes but we shouldn't fail. For gateway if do not add the check we will fail because globalEndpoints has the TLS backend. 

This code doesn't invalidate current behvaioe. It just relaxes secure gateway to be mandatory for secure backend.